tests/setup/sigterm.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/setup/sigterm.js
+ */
+'use strict'; // eslint-disable-line strict
+const tu = require('../testUtils');
+const sinon = require('sinon');
+const expect = require('chai').expect;
+const activityLogUtil = require('../../utils/activityLog');
+const jobQueueSetup = require('../../jobQueue/setup');
+const jobQueue = jobQueueSetup.jobQueue;
+
+// It starts app context
+require('../../index');
+
+describe('SIGTERM Signal handling', () => {
```

Comment:
can you think about how to confirm in tests that a new job cannot be enqueued? it would require multiple processes to be running (or at least simulated). think about it.